### DISCUSSION TITLE
tensorflow与h5py, numpy版本不兼容

### DISCUSSION THREAD

**Discussion by Chengyq234** [physics-data/faq#166] [Q&A] [answered]
Mac系统，tensorflow2.6.0要求h5py~=3.1.0 并且numpy~=1.19.2，但是h5py3.1.0要求numpy~=1.19.5，于是numpy就不兼容了怎么办

**Comment by jiegec** [comment, marked as answer]
尝试降级 h5py

**Reply by jiegec** [reply]
我试了一下：

tensorflow==2.6.0
h5py==3.1.0
numpy==1.19.5

不会产生冲突

**Reply by jiegec** [reply]
你可以试试 pip3 install tensorflow==2.6.0 h5py==3.1.0 numpy==1.19.5

**Reply by Chengyq234** [reply]
谢谢助教！我现在遇到的问题是，无法import keras 或者 tensorflow, 报错No module named 'keras'，用conda装了新环境尝试依然如此

**Reply by jiegec** [reply]
请保证你安装的python环境和你使用的python环境是同一个。

**Reply by Chengyq234** [reply]
解决了python环境的问题！现在的报错是
（1）RuntimeError: module compiled against API version 0xe but this version of numpy is 0xd；
（2）File "/usr/local/lib/python3.9/site-packages/keras/mixed_precision/loss_scale_optimizer.py", line 1180, in <module>
    mixed_precision._register_wrapper_optimizer_cls(optimizer_v2.OptimizerV2,
AttributeError: module 'tensorflow.python.training.experimental.mixed_precision' has no attribute '_register_wrapper_optimizer_cls'
其实最开始就是这个报错，我以为是版本兼容问题。请问有可能当面找助教哥哥配环境吗qwq

**Reply by jiegec** [reply]
查了一下，应该要保证tf版本和keras版本兼容。

**Reply by jiegec** [reply]
https://stackoverflow.com/questions/66178738/attributeerror-module-tensorflow-python-training-experimental-mixed-precision
